llama_stack/distribution/tracing.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
consider looking at simple lightweight spanning frameworks instead of building this.